File: bindings/csharp/Umicp.Core/Types/PayloadType.cs
using System.Text.Json.Serialization;

namespace Umicp.Core.Types;

/// <summary>
/// Payload data types for UMICP messages
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum PayloadType
{
    /// <summary>
    /// Vector data type
    /// </summary>
    Vector,

    /// <summary>
    /// Text data type
    /// </summary>
    Text,

    /// <summary>
    /// Metadata data type
    /// </summary>
    Metadata,

    /// <summary>
    /// Binary data type
    /// </summary>
    Binary,

    /// <summary>
    /// JSON data type
    /// </summary>
    Json,

    /// <summary>
    /// Matrix data type
    /// </summary>
    Matrix
}

File: bindings/kotlin/src/main/kotlin/com/hivellm/umicp/types/PayloadType.kt
package com.hivellm.umicp.types

import kotlinx.serialization.Serializable

/**
 * UMICP Payload Types
 *
 * Defines the type of data contained in a message payload.
 */
@Serializable
enum class PayloadType {
    /** Vector/array data (e.g., embeddings, tensors) */
    VECTOR,

    /** Text/string data */
    TEXT,

    /** Metadata (key-value pairs) */
    METADATA,

    /** Binary data */
    BINARY;

    companion object {
        /**
         * Convert from string value
         */
        fun fromString(value: String): PayloadType? =
            entries.find { it.name.equals(value, ignoreCase = true) }
    }
}

